WordPress/gutenberg
# Site Editor: Default to showing status slug in sidebar

## What?
Default to showing status slug to better handle custom post status types. 

<img width="659" alt="image" src="https://github.com/WordPress/gutenberg/assets/1072756/c5006e28-aaf4-452f-adf2-d57b27e608ba">


## Why?
If a custom post status is added, we currently show an empty status field. This PR improves that state slightly by showing the status slug instead. Next step is adding status types in status popover.

## Testing Instructions
1. Add a custom post status via code or a plugin [like this](https://wordpress.org/plugins/extended-post-status/)
2. Set status of a page to the new status
3. Open site editor and open page
4. Check settings sidebar and see status label
